src/dmd/astbase.d

```diff
@@ -4243,6 +4246,41 @@ struct ASTBase
         }
     }
 
+    extern (C++) final class TypeMixin : Type
+    {
+        Expressions* exps;
+
+        extern (D) this(Expressions* exps)
+        {
+            super(Tmixin);
+            this.exps = exps;
+        }
+
+        override Type syntaxCopy()
+        {
+            static Expressions* arraySyntaxCopy(Expressions* exps)
+            {
+                Expressions* a = null;
+                if (exps)
+                {
+                    a = new Expressions(exps.dim);
+                    foreach (i, e; *exps)
+                    {
+                        (*a)[i] = e ? e.syntaxCopy() : null;
+                    }
+                }
+                return a;
+            }
+
+            return new TypeMixin(arraySyntaxCopy(exps));
```

Comment:
I'm confused as to why this idiom was used here. It's just a more complicated way of saying... this:

```d
override Type syntaxCopy()
{
    Expressions* a = null;
    if (exps)
    {
        a = new Expressions(exps.dim);
        foreach (i, e; *exps)
        {
            (*a)[i] = e ? e.syntaxCopy() : null;
        }
    }
    return new TypeMixin(a);
}
```

Reply:
Because `arraySyntaxCopy` is used all over dmd, but is not available in astbase.d